polaris/ocean/viz/transect.py

```diff
@@ -0,0 +1,159 @@
+
+    Returns
+    -------
+    ds_transect : xarray.Dataset
+        The transect dataset
+    """
+
+    ds_tris = mesh_to_triangles(ds_3d_mesh)
+
+    triangle_tree = make_triangle_tree(ds_tris)
+
+    if spherical:
+        ds_transect = find_transect_cells_and_weights(
+            x, y, ds_tris, ds_3d_mesh, triangle_tree, degrees=True)
+    else:
+        ds_transect = find_planar_transect_cells_and_weights(
+            x, y, ds_tris, ds_3d_mesh, triangle_tree)
+
+    cell_indices = ds_transect.horizCellIndices
+    mask = ds_3d_mesh.maxLevelCell.isel(nCells=cell_indices) > 0
+    ds_transect = ds_transect.isel(nSegments=mask)
+
+    ds_transect = find_transect_levels_and_weights(
+        ds_transect, ds_3d_mesh.layerThickness,
+        ds_3d_mesh.bottomDepth, ds_3d_mesh.maxLevelCell - 1)
+
+    if 'landIceFraction' in ds_3d_mesh:
+        interp_cell_indices = ds_transect.interpHorizCellIndices
+        interp_cell_weights = ds_transect.interpHorizCellWeights
+        land_ice_fraction = ds_3d_mesh.landIceFraction.isel(
+            nCells=interp_cell_indices)
+        land_ice_fraction = (land_ice_fraction * interp_cell_weights).sum(
+            dim='nHorizWeights')
+        ds_transect['landIceFraction'] = land_ice_fraction
+
+    ds_transect['x'] = ds_transect.dNode.isel(
+        nSegments=ds_transect.segmentIndices,
+        nHorizBounds=ds_transect.nodeHorizBoundsIndices)
+
+    ds_transect['z'] = ds_transect.zTransectNode
+
+    ds_transect.compute()
+
+    return ds_transect
+
+
+def plot_transect(ds_transect, mpas_field, out_filename, title,
+                  colorbar_label=None, cmap=None, figsize=(12, 6), dpi=200):
+    """
+    plot a transect showing the field on the MPAS-Ocean mesh and save to a file
+
+    Parameters
+    ----------
+    ds_transect : xarray.Dataset
+        A transect dataset from
+        :py:func:`polaris.ocean.viz.compute_transect()`
+
+    mpas_field : xarray.DataArray
+        The MPAS-Ocean 3D field to plot
+
+    out_filename : str
+        The png file to write out to
+
+    title : str
+        The title of the plot
+
+    colorbar_label : str, optional
+        The colorbar label
+
+    cmap : str, optional
+        The name of a colormap to use
+
+    figsize : tuple, optional
+        The size of the figure in inches
+
+    dpi : int, optional
+        The dots per inch of the image
+
+    """
+    transect_field = interp_mpas_to_transect_triangles(ds_transect,
+                                                       mpas_field)
+
+    x_outline, z_outline = get_outline_segments(ds_transect)
+    x_outline = 1e-3 * x_outline
+
+    tri_mask = np.logical_not(transect_field.notnull().values)
+
+    triangulation_args = _get_ds_triangulation_args(ds_transect)
+
+    triangulation_args['mask'] = tri_mask
+
+    tris = Triangulation(**triangulation_args)
+    plt.figure(figsize=figsize)
+    plt.tripcolor(tris, facecolors=transect_field.values, shading='flat',
+                  cmap=cmap)
```

Comment:
```suggestion
                  cmap=cmap, vmin=vmin, vmax=vmax)
```
Can we either add `vmin` and `vmax` or `tripcolor_args` as arguments to `plot_transect`? I often use vmin and vmax to make sure that all the time slices have the same range, or to make sure that cmo.balance is symmetric about 0, for example.

Reply:
Oh, absolutely!  I don't know how I missed that!

Reply:
I have added this in the latest update.